File: TeamCode/src/org/murraybridgebunyips/bunyipslib/tasks/bases/BunyipsRunOnceTask.java
package org.murraybridgebunyips.bunyipslib.tasks.bases;

import androidx.annotation.NonNull;

import org.murraybridgebunyips.bunyipslib.BunyipsOpMode;
import org.murraybridgebunyips.bunyipslib.BunyipsSubsystem;

/**
 * RunOnceTask interface with BunyipsOpMode dependency injection.
 */
public abstract class BunyipsRunOnceTask extends RunOnceTask {
    protected final BunyipsOpMode opMode;

    protected BunyipsRunOnceTask(BunyipsOpMode opMode) {
        this.opMode = opMode;
    }

    protected BunyipsRunOnceTask(BunyipsOpMode opMode, @NonNull BunyipsSubsystem dependencySubsystem, boolean shouldOverrideConflictingTasks) {
        super(dependencySubsystem, shouldOverrideConflictingTasks);
        this.opMode = opMode;
    }
}

File: TeamCode/src/org/murraybridgebunyips/bunyipslib/tasks/bases/BunyipsRunOnceTask.kt
package org.murraybridgebunyips.bunyipslib.tasks.bases

import org.murraybridgebunyips.bunyipslib.BunyipsOpMode
import org.murraybridgebunyips.bunyipslib.BunyipsSubsystem

/**
 * RunOnceTask interface with BunyipsOpMode dependency injection.
 */
abstract class BunyipsRunOnceTask(open val opMode: BunyipsOpMode) : RunOnceTask() {
    constructor(
        opMode: BunyipsOpMode,
        dependencySubsystem: BunyipsSubsystem,
        shouldOverrideConflictingTasks: Boolean
    ) : this(opMode) {
        addDependency(dependencySubsystem)
        overrideOnConflict = shouldOverrideConflictingTasks
    }
}
